packages/react/src/components/Dropdown/Dropdown.base.tsx

```diff
@@ -163,21 +169,30 @@ export const DropdownBase: React.FunctionComponent<IDropdownProps> = React.forwa
 
     const rootRef = React.useRef<HTMLDivElement>(null);
     const mergedRootRef = useMergedRefs(forwardedRef, rootRef);
-
-    const responsiveMode = useResponsiveMode(rootRef);
     const [selectedIndices, setSelectedIndices] = useSelectedItemsState(props);
 
-    return (
-      <DropdownInternal
-        {...(props as Omit<IDropdownProps, 'ref'>)}
-        responsiveMode={responsiveMode}
-        hoisted={{ rootRef: mergedRootRef, selectedIndices, setSelectedIndices }}
-      />
-    );
+    const dropdownInternalProps = {
```

Comment:
Thanks for noticing this issue! I don't think it's necessary to use a wrapper here just to avoid a conditional hook call. Instead I'd suggest reverting most of this change, then updating the `useResponsiveMode` hook to take a second (optional) parameter for the responsive mode from props.

Reply:
Thank you for your review @ecraig12345. 

Using wrapper makes it possible to avoid listening for `resize` events, and avoid other hook calls inside `useResponsiveMode` hook itself. But I agree that adding an optional param to `useResponsiveMode` hook makes sense since there are other parts of the code that uses the `useResponsiveMode` hook and have a `responsiveMode` prop. We probably don't want to write wrapper components for all of them.

I've made some changes, please see https://github.com/microsoft/fluentui/pull/17526/commits/05bda9811eb341a19152684e8b15064031fc0c14